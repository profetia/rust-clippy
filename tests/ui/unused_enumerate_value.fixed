#![warn(clippy::unused_enumerate_value)]

fn main() {
    let mut array = [1, 2, 3];
    for index in 0..array.len() {
        //~^ unused_enumerate_value
        todo!();
    }

    let my_iter = vec![1, 2, 3].into_iter();
    for index in 0..my_iter.len() {
        //~^ unused_enumerate_value
        todo!();
    }

    let another_iter = vec![1, 2, 3].into_iter();
    for (index, _) in another_iter.enumerate().map(|(index, x)| (index, x + 1)) {
        todo!();
    }
}
