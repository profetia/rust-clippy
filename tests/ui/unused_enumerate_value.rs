#![warn(clippy::unused_enumerate_value)]

fn main() {
    let mut array = [1, 2, 3];
    for (index, _) in array.iter_mut().enumerate() {
        //~^ unused_enumerate_value
        todo!();
    }

    let my_iter = vec![1, 2, 3].into_iter();
    for (index, _) in my_iter.enumerate() {
        //~^ unused_enumerate_value
        todo!();
    }

    let another_iter = vec![1, 2, 3].into_iter();
    for (index, _) in another_iter.enumerate().map(|(index, x)| (index, x + 1)) {
        todo!();
    }
}
