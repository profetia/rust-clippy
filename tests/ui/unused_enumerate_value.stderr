error: you seem to use `.enumerate()` and immediately discard the value
  --> tests/ui/unused_enumerate_value.rs:5:23
   |
LL |     for (index, _) in array.iter_mut().enumerate() {
   |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: `-D clippy::unused-enumerate-value` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(clippy::unused_enumerate_value)]`
help: replace `array.iter_mut().enumerate()` with `0..array.len()`
   |
LL -     for (index, _) in array.iter_mut().enumerate() {
LL +     for index in 0..array.len() {
   |

error: you seem to use `.enumerate()` and immediately discard the value
  --> tests/ui/unused_enumerate_value.rs:11:23
   |
LL |     for (index, _) in my_iter.enumerate() {
   |                       ^^^^^^^^^^^^^^^^^^^
   |
help: replace `my_iter.enumerate()` with `0..my_iter.len()`
   |
LL -     for (index, _) in my_iter.enumerate() {
LL +     for index in 0..my_iter.len() {
   |

error: aborting due to 2 previous errors

